src/main/java/core/basesyntax/strategy/ReturnOperation.java

```
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.db.Storage;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
```

The handler does not validate its input. You must check that `fruitTransaction` is non-null before using it and throw an informative RuntimeException if it is null (requirements: "Handlers should validate inputs (non-null transaction, fruit non-null/non-empty, and non-negative quantity)"). Add a null-check at the start of `apply` to fail fast. Current location: method signature at line 8.